pycsob/client.py

```diff
@@ -17,6 +20,202 @@ def send(self, request, **kwargs):
         return super(HTTPAdapter, self).send(request, **kwargs)
 
 
+class ConvertMixin:
+    """Convert instance into ordered dict."""
+    
+    def _to_camel_case(self, name: str) -> str:
+        """Convert name to camel case form."""
+        parts = name.split("_")
+        return parts[0] + "".join(i.title() for i in parts[1:])
+
+    def _fields(self) -> Tuple[str, ...]:
+        return (i.name for i in fields(self))
+
+    def to_dict(self) -> Dict[str, Union[str, Dict]]:
+        data = []
+        for name in self._fields():
+            value = getattr(self, name)
+            if value in conf.EMPTY_VALUES:
+                continue
+            if hasattr(value, 'to_dict'):
+                value = value.to_dict()
+            data.append((self._to_camel_case(name), getattr(self, f"_format_{name}", lambda v: v)(value)))
+        return OrderedDict(data)
+
+
+@dataclass
+class CartItem(ConvertMixin):
+    """Cart item for creating card payment."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Basic-methods#item---cart-item-object-cart-
+
+    name: str
+    quantity: int
+    amount: int
+    description: Optional[str] = None
+    
+    def _format_name(self, value: str) -> str:
+        return value[:20]
+    
+    def _format_description(self, value: str) -> str:
+        return value[:40]
+
+
+@dataclass
+class CustomerAccount(ConvertMixin):
+    """Customer account data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customeraccount-data-
+
+    created_at: Optional[str] = None
+    changed_at: Optional[str] = None
+    changed_pwd_at: Optional[str] = None
+    order_history: Optional[int] = None
+    payments_day: Optional[int] = None
+    payments_year: Optional[int] = None
+    one_click_adds: Optional[int] = None
+    suspicious: Optional[bool] = None
+
+
+@unique
+class CustomerLoginType(Enum):
+    """Type of customer login."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customerlogin-data-
+
+    GUEST = "guest"
+    ACCOUNT = "account"
+    FEDERATED = "federated"
+    ISSUER = "issuer"
+    THIRDPARTY = "thirdparty"
+    FIDO = "fido"
+    FIDO_SIGNED = "fido_signed"
+    API = "api"
```

Comment:
It would be handy to add the `__str__` function into `Enum` classes.

```python
def __str__(self):
    return str(self.value)
```
```
>>> str(CustomerLoginType.GUEST)
'guest'
```

Reply:
OK. But there is no string conversion of value when storing it into `OrderedDict` as payload and consequently there is an exception raised when JSON serialization (`TypeError: Object of type CustomerLoginType is not JSON serializable`).